upload.go

```diff
@@ -0,0 +1,85 @@
+package attache
+
+import (
+	"bytes"
+	"fmt"
+	"image"
+	_ "image/gif"
+	_ "image/jpeg"
+	_ "image/png"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rwcarlsen/goexif/exif"
+)
+
+func (s Server) handleUpload(w http.ResponseWriter, r *http.Request) (result uploadResponse, err error) {
+	stream := &bytes.Buffer{}
+	_, err = io.Copy(stream, r.Body)
+	if err != nil {
+		return result, errors.Wrapf(err, "copy")
+	}
+	defer r.Body.Close()
+
+	file := bytes.NewReader(stream.Bytes())
+	fileType := http.DetectContentType(stream.Bytes())
+	fileLen := stream.Len()
+
+	filePath, err := s.Storage.Upload(file, fileType)
+	if err != nil {
+		return result, errors.Wrapf(err, "upload")
+	}
+
+	fileMeta := meta(file, fileType)
+
+	result = uploadResponse{
+		Path:        filePath,
+		ContentType: fileType,
+		Bytes:       fileLen,
+		Meta:        fileMeta,
+	}
+
+	return result, nil
+}
+
+func meta(file *bytes.Reader, fileType string) uploadMeta {
+	fileMeta := uploadMeta{DateTime: "", LatLong: "", Geometry: ""}
+
+	if strings.Contains(fileType, "image") {
+		imageMeta(file, &fileMeta)
+	}
+
+	return fileMeta
+}
+
+func imageMeta(file *bytes.Reader, fileMeta *uploadMeta) {
+	file.Seek(0, 0)
+
+	x, err := exif.Decode(file)
+	if err != nil {
+		log.Println(err.Error())
+	} else {
+		xDateTime, xerr := x.DateTime()
+		if xerr != nil {
+			log.Println(xerr.Error())
+		}
+		fileMeta.DateTime = xDateTime.String()
+
+		xLat, xLong, xerr := x.LatLong()
+		if xerr != nil {
+			log.Println(xerr.Error())
+		}
```

Comment:
let's skip these `xerr`; no values

Reply:
Need to understand more about when we capture and bubble up errors in the idiomatic Go.

Reply:
Will skip the `xerr` values. In this case, I'll not return any error messages actually, rendering the next comment https://github.com/winston/attache-lambda/pull/2#discussion_r157371999 somewhat invalid?

Assumption is that the extraction of file meta data can fail horribly, because the upload could be an `image` but not a `photo` (which has EXIF) and so would fail at EXIF decode. But code wise, it should continue to try and figure out as much meta as possible (moving on to finding out dimensions) instead of failing/break and return.

Reply:
> Will skip the xerr values. In this case, I'll not return any error messages actually

👌 

> rendering the next comment #2 (comment) somewhat invalid?

not really. that comment is a recommendation that returning error is a valid pattern. 

i.e. if the underlying lib you use returns error, you should not try to handle it so deep inside; just return the wrapped error

Reply:
> it should continue to try and figure out as much meta as possible (moving on to finding out dimensions) instead of failing/break and return

totally 💯 👍